pkgs/development/python-modules/visions/default.nix

```diff
@@ -0,0 +1,61 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, pytestCheckHook
+, attrs
+, imagehash
+, matplotlib
+, multimethod
+, networkx
+, numpy
+, pandas
+, pillow
+, pydot
+, pygraphviz
+, shapely
+}:
+let
+  optional-dependencies = {
+    type-geometry = [ shapely ];
+    type-image-path = [ imagehash pillow ];
+    plotting = [ matplotlib pydot pygraphviz ];
+  };
+in
+buildPythonPackage rec {
+  pname = "visions";
+  version = "0.7.6";
+  disabled = pythonOlder "3.8";
+
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "dylan-profiler";
+    repo = "visions";
+    rev = "5fe9dd0c2a5ada0162a005c880bac5296686a5aa"; # no tagged release for 0.7.6
+    hash = "sha256-SZzDXm+faAvrfSOT0fwwAf9IH7upNybwKxbjw1CrHj8=";
+  };
+
+  propagatedBuildInputs = [ attrs multimethod networkx numpy pandas ];
+
+  nativeCheckInputs =
+    [ pytestCheckHook imagehash ]
+    ++ lib.flatten (builtins.attrValues passthru.optional-dependencies);
```

Comment:
same here, use final.passthru

Reply:
I'm not sure this is the appropriate use of `final`, seems `lib.fix` is what's used under the hood and also what packages like [`sqlalchemy`](https://github.com/NixOS/nixpkgs/blob/274d22e4c269a3ebc4421529215f3b972db24c84/pkgs/development/python-modules/sqlalchemy/default.nix#L70) use.

Reply:
Sounds good.